backend/app/app/api/api_v1/endpoints/ue_movement.py

```diff
@@ -1,507 +1,318 @@
+
+            await asyncio.sleep(1)
+
+
+async def location_notification(
+    ue: UE, old_cell_id: Optional[int], current_cell_id: Optional[int]
+):
+    db_mongo = client.fastapi
+
+    subscriptions = list(
+        db_mongo["MonitoringEvent"].find(
+            {"supi": str(ue.supi)},
+        ),
+    )
+
+    for doc in subscriptions:
+        sub = doc["subscription"]
+        sub_validate_time = tools.check_expiration_time(
+            expire_time=sub.get("monitorExpireTime")
+        )
+
+        sub_validate_number_of_reports = tools.check_numberOfReports(
+            sub.get("maximumNumberOfReports")
+        )
+
+        if not sub_validate_time or not sub_validate_number_of_reports:
+            crud_mongo.delete_by_uuid(db_mongo, "MonitoringEvent", doc.get("_id"))
+            continue
+
+        if sub.get("maximumNumberOfReports") is not None:
+            sub["maximumNumberOfReports"] = sub["maximumNumberOfReports"] - 1
+            doc["subscription"] = sub
+            crud_mongo.update(db_mongo, "MonitoringEvent", doc.get("_id"), doc)
+
+        monitoringType = sub["monitoringType"]
+
+        if monitoringType == MonitoringType.LOCATION_REPORTING:
+            await handle_location_report_callback(sub, ue)
+
+        elif monitoringType == MonitoringType.LOSS_OF_CONNECTIVITY:
+            asyncio.create_task(
+                handle_loss_connectivity_callback(sub, ue, old_cell_id, current_cell_id)
+            )
+
+        elif monitoringType == MonitoringType.UE_REACHABILITY:
+            await handle_ue_reachability_callback(sub, ue, old_cell_id, current_cell_id)
+
+
+async def handle_location_report_callback(location_reporting_sub, ue: UE):
+    db_mongo = client.fastapi
+    logging.info(
+        "Attempting to send the callback to %d",
+        location_reporting_sub.get("notificationDestination"),
+    )
+
+    notification = MonitoringNotification(
+        subscription=location_reporting_sub.get("self"),
+        monitoringEventReports=[
+            MonitoringEventReport(
+                externalId=ue.external_identifier,
+                monitoringType=MonitoringType.LOCATION_REPORTING,
+                locationInfo=LocationInfo(
+                    cellId=ue.Cell_id,
+                    enodeBId=None,
+                    geographicArea=Point(
+                        shape=SupportedGADShapes.POINT,
+                        point=GeographicalCoordinates(
+                            lat=ue.latitude,
+                            lon=ue.longitude,
+                        ),
+                    ),
+                ),
+            )
+        ],
+    )
+
+    await notification_responder.send_notification(
+        location_reporting_sub.get("notificationDestination"), notification
+    )
+
+
+async def handle_loss_connectivity_callback(
+    loss_of_connectivity_sub,
+    ue: UE,
+    old_cell_id: Optional[int],
+    current_cell_id: Optional[int],
+):
+    if old_cell_id and not current_cell_id:
```

Comment:
```suggestion
    if not old_cell_id or current_cell_id:
        return
```

Invert the if into a check guard to reduce the indentation